lib/src/main/kotlin/com/lemonappdev/konsist/api/architecture/KoArchitectureAssertion.kt

```diff
@@ -13,29 +13,103 @@ interface KoArchitectureAssertion {
      * @param dependencies The function literal with receiver [DependencyRules] that allows configuring the dependencies
      * of the architecture.
      */
```

Comment:
I added separate method for each case (was 4 methods, now is 8 methods) to enable writing code as before.

After changes we have few parameters, so `dependencies` lambda parameter should be the last to enable writing code like this:
<img width="360" alt="Pasted Graphic 2" src="https://github.com/LemonAppDev/konsist/assets/111683562/1940f98f-e35c-4eb3-bad4-ea419bfcf350">

But then we cannot write such code
<img width="443" alt="Pasted Graphic 1" src="https://github.com/LemonAppDev/konsist/assets/111683562/094229ad-0510-4033-87b8-aad7cd713d65">
because we should write `scope.assertArchitecture(dependencies = architecture)` (use name argument).

Now 
- when we pass only `dependencies` parameter we may do this as before (like in the above images)
- when we pass any other parameter (`testName` or `additionalMessage`) and `architecture` as parameter, then we have to use named arguments﻿
    <img width="641" alt="image" src="https://github.com/LemonAppDev/konsist/assets/111683562/b84905e6-4b02-464f-997d-4657ebf95309">
- when we pass any other parameter (`testName` or `additionalMessage`) and `architecture` as lambda code, then we do this as always:
    <img width="437" alt="image" src="https://github.com/LemonAppDev/konsist/assets/111683562/9ecd6820-ffde-4624-9df6-1a81fcad26a7">